rust/agents/scraper/src/scraper/mod.rs

```diff
@@ -261,9 +390,22 @@ impl SqlOutboxScraper {
                     .collect::<Vec<_>>(),
             ) {
                 ListValidity::Valid => {
-                    // Difference 3
-                    let max_leaf_index_of_batch = self.store_messages(&sorted_messages).await?;
+                    // transaction (database_id, timestamp) by transaction hash
+                    let txns: HashMap<H256, (i64, TimeDateTime)> = self
+                        .ensure_blocks_and_txns(
+                            sorted_messages
+                                .iter()
+                                .map(|(_, meta)| meta)
+                                .chain(deliveries.iter().map(|d| &d.meta)),
+                        )
+                        .await?
+                        .collect();
+
+                    let max_leaf_index_of_batch =
+                        self.store_messages(&sorted_messages, &txns).await?;
                     stored_messages.inc_by(sorted_messages.len() as u64);
+                    self.record_deliveries(&deliveries, &txns).await?;
+                    stored_deliveries.inc_by(deliveries.len() as u64);
```

Comment:
This gives an accurate number because of the definition of the metric. I had to change it because it was incorrectly used elsewhere as well so instead of changing how we record the metric to could how many are stored in the database, it tracks how many are stored into the database. Sorry, english is stupid here, but it is meant to track the number of store operations not the final count in the db.

```rs
/// Events stored into AbacusDB (label values differentiate checkpoints vs.
/// messages)
///
/// Labels:
/// - `data_type`: the data the indexer is recording. E.g. `messages` or `gas_payments`.
/// - `chain`: Chain the indexer is collecting data from.
pub stored_events: IntCounterVec,
```